README.rst

```diff
@@ -134,6 +134,24 @@ More examples:
     with open('s3://bucket/key.txt', 'wb', transport_params=transport_params) as fout:
         fout.write(b'here we stand')
 
+The ``open`` function also accepts file-like objects.
+This is useful when you already have an open file, and would like to transparently decompress it.
+
+.. code-block:: python
+
+    >>> import io
+    >>> filepath = 'smart_open/tests/test_data/1984.txt.gz'
+    >>> with io.open(filepath, 'rb') as open_file:
```

Comment:
I propose demonstrating on `BytesIO`, to make it clearer we're talking about arbitrary wrapping binary streams, rather than just (local) files.